bastionzero/service/targets/dbauthconfig/dbauthconfig.go

```diff
@@ -0,0 +1,13 @@
+package dbauthconfig
+
+// DatabaseAuthenticationConfig defines a database authentication configuration supported
+// by BastionZero. When using a non-null DatabaseAuthenticationConfig in a request, it
+// is recommended that the supported configurations are retrieved from a GET request to
+// /api/v2/targets/database/supported-database-configs and then one of the returned
+// configurations is used in any subsequent create or update request as needed.
+type DatabaseAuthenticationConfig struct {
+	AuthenticationType   *string `json:"authenticationType"`
+	CloudServiceProvider *string `json:"cloudServiceProvider"`
+	Database             *string `json:"database"`
+	Label                *string `json:"label"`
```

Comment:
These are all optional I think (inferred by pointer type). So can you add `omitempty` in the `json` tag, so that the field is not serialized if unset by client

Reply:
Got it. I figured I forgot at least one thing somewhere. 😀

Reply:
b574bc2d38a8307385ccf82081f6a1253b339e7c